python/ycm/client/command_request.py

```diff
@@ -69,9 +71,50 @@ def RunPostCommandActionsIfNeeded( self ):
         vimsupport.JumpToLocation( self._response[ 'filepath' ],
                                     self._response[ 'line_num' ],
                                     self._response[ 'column_num' ] )
-    elif 'message' in self._response:
-      vimsupport.EchoText( self._response['message'] )
+    elif self._is_fixit_command:
+      if not len( self._response[ 'fixits' ] ):
+        vimsupport.EchoText( "No fixits found for current line" )
+      else:
+        fixit = self._response[ 'fixits' ][ 0 ]
+
+        # We need to track the difference in length, but ensuring we apply fixes
+        # in ascending order of insertion point.
+        fixit[ 'chunks' ].sort( key = lambda chunk:  (
+          str(chunk[ 'range' ][ 'start' ][ 'line_num' ])
+          + ','
+          + str(chunk[ 'range' ][ 'start' ][ 'column_num' ])
+        ))
+
+        # Remember the line number we're processing. -ve line number means we
+        # haven't processed any lines yet (by nature of being not equal to any
+        # real line number).
+        last_line = -1
+
+        # Counter of changes applied, so the user has a mental picture of the
+        # undo history this change is creating.
+        fixed = 0
```

Comment:
sure thing, will update. i've a habit of going for the whole brevity-single-noun thing.